coveralls/reporter.py

```diff
@@ -183,9 +189,13 @@ def get_arcs(analysis):
     def parse_file(self, cu, analysis):
         """Generate data for single file."""
         filename = cu.relative_filename()
+
         # ensure results are properly merged between platforms
         posix_filename = filename.replace(os.path.sep, '/')
 
+        if self.base_dir and posix_filename.startswith(self.base_dir):
```

Comment:
```suggestion
        if posix_filename.startswith(self.base_dir):
```

See above suggestion for making `self.base_dir` always a string to make this work.

Reply:
I left the test here because `startswith('')` is always true and results in unnecessary work in the common case of no base dir